### DISCUSSION TITLE
failed configuration when attempting install

### DISCUSSION THREAD

**Discussion by phildrage** [Blackymas/NSPanel_HA_Blueprint#711] [Q&A]
[logs_nspanel_compile.txt](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/11302163/logs_nspanel_compile.txt)

When trying to download the configuration to the nspanel I get the error as shown in the logs. Can anyone suggest why this might be?

**Comment by edwardtfn** [comment]
I cannot open your file now from my phone, but could you please share your ESPHome yaml? Please make sure to hide any private info like wifi credentials.

**Reply by phildrage** [reply]
turns out my version of ESPHome was depricated!! New version seems to sort the issue. Gone through the process just fine now though, and  i'm getting no display on the nspanel at all. Reflashed the firmware successfully again, but no display at all. Wondering if I have a dud panel.

**Reply by phildrage** [reply]
Right, Got it working. I'd missed the nextion update url, I had put the IP address in of the server, but hadn't noticed that the rest had 'us' at the end, and I'm using the 'eu' file! Onto the next step! looking good so far, Home assistant has found it at added it as a device.

**Reply by edwardtfn** [reply]
Can you now see the info from Home Assistant in your panel? Or still missing something?

**Reply by phildrage** [reply]
nope totally stuck now. couldnt update the tft file, so tried the updated yaml file with the additional lines in, and got it going. Update started but something happened about 5 mins in and the panel rebooted. (maybe loose connection). Panel now has 'system data error' sideways on the screen and I can't seem to get it updated!

**Reply by edwardtfn** [reply]
Try to upload the `nspanel_blank.tft` first. After that is installed, you don't need this additional lines anymore and will be able to use the latest Nextion library to upload the bigger TFT.
And always use a local www server ( like the one embedded into Home Assistant).

**Reply by phildrage** [reply]
I can flash the esp32 but the screen always says system data error, and i never get a progress bar when trying to update the tft. The buttons seem to respond in HA so it is connected in some way...

**Reply by edwardtfn** [reply]
> The buttons seem to respond in HA so it is connected in some way

Ok, this is an important info. If your buttons are responding, that means ESPHome is working as the Blueprint.

Now you have to focus on transferring the TFT files to your panel.

Few questions so I can understand where you are in this process and how I would be able to support:
1. Are you trying to transfer the tft files from GitHub directly to your panel or are you hosting the files in a local www server and trying to upload from there?
2. Have you tried the `nspanel_blank.tft`? That file is only 258kb (compared to more than 6mb on the `nspanel_eu.tft`) so let's focus on that first and once you can transfer that then we try to repeat the same process for `nspanel_eu.tft`.
3. Could you please share the ESPHome yaml you are using? I know you already solved the ESPHome part, but there is an older Nextion library which manages the first TFT upload better and I will use your current ESPHome to propose some changes to use temporarily for this first transfer. Please remember to remove any private info (WiFi credentials, etc) before sharing.

**Reply by phildrage** [reply]
I'm hosting locally,

I'm getting no HMI response when trying to upload _blank,

`substitutions:

###### CHANGE ME START ######

  device_name: "kitchennspanel" 
  wifi_ssid: !secret wifi_ssid
  wifi_password: !secret wifi_password

#  nextion_update_url: "http://192.168.1.42:8123/local/nspanel_eu.tft" # URL to local tft File
  nextion_update_url: "http://192.168.1.42:8123/local/nspanel_blank.tft" # URL to local tft File
#  nextion_update_url: "https://raw.githubusercontent.com/Blackymas/NSPanel_HA_Blueprint/main/nspanel_eu.tft" # URL to Github

##### CHANGE ME END #####


###### USE THIS ONLY FOR YOUR FIRST TFT UPLOAD, ONCE IT WORKED, REMOVE THOSE LINES ######
external_components:
  - source: github://pr#2956
    components: [nextion]
    refresh: 1h
###############################################################################


##### DO NOT CHANGE ANYTHING! #####

packages:
  ##### download esphome code from Github
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: dev
    files: [nspanel_esphome.yaml]
    refresh: 300s

##### DO NOT CHANGE ANYTHING! #####`

**Reply by phildrage** [reply]
I have some progress, I bypassed the esp32 and connected directly to the tft using the rx/tx pins as shown in issue #166, (note the uart connection is rx-rx, tx-tx which caught me out for a while!) and flashed the blank screen successfully. I've now changed to the larger tft file and removed the additional lines in the yaml and I'll let you know how i get on! looking good so far.

**Reply by phildrage** [reply]
I now have data from HA, but I'm stuck on the loading screen. I'm following through your guide #671 and can't seem to find any issues. The panel is called 'kitchennspanel' in esphome and automations. I also have set up accuweather and tried both default and the accuweather setting in the automation. The hardware buttons are triggering an action correctly, but still stuck on loading screen. The display brightness responds correctly too. Is there anything else I can check? on boot, the screen indicates tft:3.2.2, espHome:3.3, blueprint:3.3

**Reply by edwardtfn** [reply]
You have different versions between the parts. I would suggest to use all from v3.2.2 as this is the latest released (stable?) version and will make easier as you don't have to update your TFT again.

So, in order to do this, change your ESPHome yaml. In the `remote_package` part you have it pointing to `dev` with `ref: dev`. Please change that to `ref: main`.
Also, download the Blueprint from the main branch: https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/nspanel_blueprint.yaml

**Reply by phildrage** [reply]
It works! superb! Thanks so much for the assistance, its much appreciated! It could be the versions, but also (in case it helps someone else)

- changed the name in esphome as well as the automation to 'nspanel' incase 'kitchennspanel' was too long. Deleted device from HA and reflashed and re-updated the TFT, HA seemed to pick up the old name again which was strange, I thought it could be because its picked up the same IP, so i forced a fixed IP to a different one, no change.
- instead of fighting it, changed the names back to 'kitchennspanel', and it all sprang into life!!!!

**Reply by phildrage** [reply]
failed again. We had a power cut and the nspanel rebooted along with the whole house. On reboot the nspanel is stuck on 'initialising'. I'm trying to update everything to the versions you suggested but no joy. I've deleted the blueprint, automations, device and rebooted, reflashed new firmware to the nspanel, and now its stuck. I have the home assistant screen with initialising, please wait, but never gets further. Home assistant doesn't see it as a device either so i can't select it in the automation.

**Reply by edwardtfn** [reply]
> Home assistant doesn't see it as a device either so i can't select it in the automation.

OK, this is where we have to focus then.
I believe you are seen what we call the boot page. Can you see if you are getting a valid ip (it should be shown in the top of that screen)?
On Home Assistant, go to Settings > Devices and Services, click in Add integration and select ESPHome, then try to add using the device name or the device ip.

**Reply by edwardtfn** [reply]
By the way, if you are with boot page you should have a reset button in your screen. Have you tried that?
Anything in your ESPHome logs?

**Reply by phildrage** [reply]
no IP adress, doesn't appear on my router as a device, and nothing in esphome logs

**Reply by phildrage** [reply]
Yes, I've tried the 'reboot' button but comes back to the same page
